adapter/internal/discovery/xds/server.go

```diff
@@ -295,6 +297,38 @@ func DeleteAPI(vhost, apiName, version string) error {
 	return nil
 }
 
+// UndeployAPI undeploys the APIs from the provided set of enviroments. If the complete set of provided environments
+// and already deployed environments are the same the apim
+func UndeployAPI(vhost, apiName, version string, undeployedEnvs []string) {
+	apiIdentifier := GenerateIdentifierForAPI(vhost, apiName, version)
+	existingLabels, ok := openAPIEnvoyMap[apiIdentifier]
+	if !ok {
+		logger.LoggerXds.Debugf("No API to undeploy under %s:%s", apiName, version)
+		return
+	}
+	newLabels := []string{}
+	for _, existingEnv := range existingLabels {
+		labelMatched := false
+		for _, undeployedEnv := range undeployedEnvs {
+			if undeployedEnv == existingEnv {
+				labelMatched = true
+				break
+			}
+		}
+		if !labelMatched {
+			newLabels = append(newLabels, existingEnv)
+		}
```

Comment:
We can use the existing function `arrayContains` here, and add to slice if not exists